File: Backend/MuseumAPI/Models/Painting.cs
namespace MuseumAPI.Models
{
    public class Painting
    {
        public long Id { get; set; }
        public string? Title { get; set; }
        public int CreationYear { get; set; }
        public double Height { get; set; }
        public string? Subject { get; set; }
        public string? Medium { get; set; }
        public string? Description { get; set; }
        public double Price { get; set; }

        public long? ArtistId { get; set; }

        // Hidden from the API because it's not in the DTO
        public virtual Artist? Artist { get; set; } = null!;

        public virtual long? UserId { get; set; }
        public virtual User? User { get; set; } = null!;
    }
}

File: Frontend/src/models/Painting.ts
import { Artist } from './Artist';
import { User } from './User';

export interface Painting {
    id?: number;
    title: string;
    creationYear: number;
    height: number;
    subject: string;
    medium: string;
    description?: string;
    price: number;
    artist?: Artist;
    artistId: number;

    userId?: number;
    user?: User;

    [key: string]: any;
}